media-plugins/gkrellmpc/gkrellmpc-0.1_beta10-r1.ebuild

```diff
@@ -0,0 +1,43 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit gkrellm-plugin toolchain-funcs
+
+DESCRIPTION="A gkrellm plugin to control the MPD (Music Player Daemon)"
+HOMEPAGE="http://mpd.wikia.com/wiki/Client:GKrellMPC"
+SRC_URI="http://mina.naguib.ca/dist/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="threads"
+
+RDEPEND="
+	app-admin/gkrellm:2[X]
+	net-misc/curl"
+DEPEND="${RDEPEND}"
+
+# Will open gkrellm in X11 display
+RESTRICT="test"
+
+PATCHES=( "${FILESDIR}/${PF}"-make-test.patch )
+
+src_prepare() {
+	if use threads; then
+		PATCHES+=( "${FILESDIR}/${P}"-mt.patch )
```

Comment:
It's not going to be a real issue for this, but conditional patching is to be avoided if we can help it because:
1. we don't notice if patches stop applying if not testing with that flag
2. it makes problems harder to debug (introduces more variance)

Patches should be written such that they can be applied unconditionally even if they only take effect conditionally based on build options.

Reply:
bonus: are these issues with conditional patching actually mentioned in devmanual? they should be. maybe you could add it ;)

Reply:
Thanks for pointing this out. I don't think conditional patching is really mentioned in the devmanual - I'll take a look.

Reply:
I found this in the devmanual: https://devmanual.gentoo.org/ebuild-writing/use-conditional-code/.

But I'm thinking [this page](https://devmanual.gentoo.org/ebuild-writing/misc-files/patches/index.html) could be expanded with a subsection called "conditional patching" that warns against the use of conditional patching for the reasons you gave.

Reply:
How do you feel about drafting something up?

Reply:
I'll do it!

Reply:
https://github.com/gentoo/devmanual/pull/281